libpromises/locks.c

```diff
@@ -506,7 +469,30 @@ static bool KillLockHolder(const char *lock)
 
     CloseLock(dbp);
 
-    return GracefulTerminate(lock_data.pid, lock_data.process_start_time);
+    if (GracefulTerminate(lock_data.pid, lock_data.process_start_time))
+    {
+        Log(LOG_LEVEL_INFO,
+            "Process with PID %jd killed",
+            (intmax_t) lock_data.pid);
+        return true;
+    }
+    else
+    {
+        if (errno == ESRCH)
+        {
+            Log(LOG_LEVEL_VERBOSE,
+                "Process with PID %jd has already been killed",
+                (intmax_t) lock_data.pid);
```

Comment:
Have you been able to test that this works somehow? Either an automated test or a manual way to reproduce?

Reply:
I am struggling with recreating the issue, but that might be because I am not familiar enough with CFEngine. I looked through the unit tests and there were none that utilised the locks.

Reply:
Issue has been recreated:

Previous noisy error message when process has been killed already:
```
    info: Lock expired after 1/1 minutes: lock.main.commands.action_name.action.ifelapsed.expireafter.-dev._bin_sleep_70_238_MD5=ca1a5ea5d7318d1c7ecbed87b0fef8fc
   error: Unable to kill expired process 3520 from lock lock.main.commands.action_name.action.ifelapsed.expireafter.-dev._bin_sleep_70_238_MD5=ca1a5ea5d7318d1c7ecbed87b0fef8fc (probably process not found or permission denied)
```

New non-noisy verbose message:
```
    info: Lock expired after 1/1 minutes: lock.main.commands.action_name.action.ifelapsed.expireafter.-dev._bin_sleep_70_238_MD5=ca1a5ea5d7318d1c7ecbed87b0fef8fc
 verbose: Process with PID 1660 has already been killed
 verbose: Lock successfully expired: lock.main.commands.action_name.action.ifelapsed.expireafter.-dev._bin_sleep_70_238_MD5=ca1a5ea5d7318d1c7ecbed87b0fef8fc
```

Reply:
Maybe the "Lock expired after 1/1" should be "Expiring lock after 1/1 minutes:", and "Lock successfully expired:" should be info and not verbose?